lib/active_model_serializers/adapter/attributes.rb

```diff
@@ -71,8 +76,21 @@ def relationship_value_for(association, options)
         return association.options[:virtual_value] if association.options[:virtual_value]
         return unless association.serializer && association.serializer.object
 
+        if options[:fields].is_a? Array
```

Comment:
I think you might also be able to use the `ActiveModel::Serializer::Fieldset.new(options[:fields]).fields_for(association.name)` like the JSON API adapter does